### DISCUSSION TITLE
Building a splici index for two different read lengths

### DISCUSSION THREAD

**Discussion by lawsonvt** [COMBINE-lab/alevin-fry#72] [Q&A] [answered]
Hello,

I have a pre-processed, paired-end snRNASeq dataset I want to to align in which the first read is 26bp and the second read is 91bp. How do I generate a single splici index for two different read lengths?

Thank you,

~ Mark

**Comment by rob-p** [comment, marked as answer]
Hi @lawsonvt,

  In this case, the first read is techinal (UMI and barcode) and not relevant to the index construction. You should treat 91 as your read length.

Best,
Rob

**Reply by lawsonvt** [reply]
Thanks Rob. Does this mean the reads should be aligned as single reads or still as paired?

**Reply by rob-p** [reply]
What protocol were they sequenced with? `salmon alevin` will accept a pair and just "do the right thing" if you pass in the correct protocol flag.

Best,
Rob

**Reply by lawsonvt** [reply]
"snRNA-seq libraries were obtained using 10X Genomics Chromium Single Cell 5′ v2 chemistry following nuclear dissociation"

Is that enough info?

**Reply by rob-p** [reply]
Yes; this is sufficient information.  Basically, it means you should run `salmon alevin` with the `--chromium` flag — with something like:

```
salmon alevin quant -i <index> -l ISF -1 <read1> -2 <read2> --chromium --sketch -p <num_threads> -o <output_map_dir> 
```

Then, when you perform the `generate-permit-list` step of `alevin-fry`, make sure to filter with the orientation one expects for the `5'` 10x protocol.  That is, you should pass `-d rc` to the `generate-permit-list` command, since we expect the biological read to map in the reverse complement orientation in this protocol.  Otherwise, the process should be as normal.  Let us know if you run into any other questions.

--Rob

**Comment by robertzeibich** [comment]
I ran into something similar. I have FASTQs with different read lengths:
R1: 26 R2: 124
R1: 150 R2: 150
R1: 151 R2: 151

Do I have to create different reference indices?
What should r-len be? Should I use 91, or 124, 150, and 151 instead?

I am using simpleaf (alevin-fry).

**Reply by robertzeibich** [reply]
I just had a look at the following [post](https://github.com/COMBINE-lab/alevin-fry/issues/21). Since the specific length does not seem to make a difference, I will use 91. If I am mistaken, please correct me. 

Based on the following [tutorial](https://combine-lab.github.io/alevin-fry-tutorials/2023/simpleaf-piscem/), **you recommend checking and using the exact read length of the processed dataset**.

I am still confused. 

A response would be much appreciated.
